sinks/signalfx/signalfx.go

```diff
@@ -273,6 +289,41 @@ const (
 	limitQueryValue = 200
 )
 
+func getSessionToken(client *http.Client, endpoint, email, password string) (string, error) {
+	sessionEndpoint := endpoint + sessionAddr
+
+	jsonBody := []byte(fmt.Sprintf("{\"email\": \"%s\", \"password\": \"%s\"}", email, password))
+	resp, err := client.Post(sessionEndpoint, "application/json", bytes.NewReader(jsonBody))
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusUnauthorized {
+		return "", fmt.Errorf("invalid credentials when generating session token")
+	} else if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("signalfx api returned unknown response code when generating session: %s", resp.Status)
+	}
+
+	body := &bytes.Buffer{}
+	_, err = body.ReadFrom(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	var createSessionResponse = make(map[string]interface{})
+	err = json.Unmarshal(body.Bytes(), &createSessionResponse)
+	if err != nil {
+		return "", err
+	}
+
+	accessToken, ok := createSessionResponse["accessToken"].(string)
+	if !ok {
+		return "", fmt.Errorf("No access token returned when creating session")
+	}
+	return accessToken, nil
+}
+
```

Comment:
Any reason not to cache the session token here until it expires?

Reply:
Not caching the session token seemed simpler when implementing.  An extra api call to create a session token each time we fetch tokens shouldn't add that much overhead.